mllib/src/main/scala/org/apache/spark/mllib/linalg/distributed/BlockMatrix.scala

```diff
@@ -246,4 +246,135 @@ class BlockMatrix(
     val localMat = toLocalMatrix()
     new BDM[Double](localMat.numRows, localMat.numCols, localMat.toArray)
   }
+
+  /**
+   * Forms [[MatrixBlock]]s using `rowsPerBlock` and `colsPerBlock`, which were provided in
+   * the constructor. Even if all [[MatrixBlock]]s (except the ones on the right and bottom edge)
+   * have dimensions `rowsPerBlock`x`colsPerBlock`, a new [[BlockMatrix]] will be returned.
+   */
+  def repartition(): BlockMatrix = {
+    repartition(rowsPerBlock, colsPerBlock)
+  }
+
+  /**
+   * Forms [[MatrixBlock]]s using the provided dimensions `newRowsPerBlock`x`newColsPerBlock`
+   * Even if all [[MatrixBlock]]s (except the ones on the right and bottom edge)
+   * have dimensions `newRowsPerBlock`x`newColsPerBlock`, a new [[BlockMatrix]] will be returned.
+   * The resulting number of partitions may be different than the current number of partitions.
+   *
+   * @param newRowsPerBlock Number of rows that make up each MatrixBlock.
+   * @param newColsPerBlock Number of columns that make up each MatrixBlock.
+   * @return The repartitioned BlockMatrix
+   */
+  def repartition(newRowsPerBlock: Int, newColsPerBlock: Int): BlockMatrix = {
+    repartition(newRowsPerBlock, newColsPerBlock, blocks.partitions.length)
+  }
+
+  /**
+   * Forms [[MatrixBlock]]s using the provided dimensions `newRowsPerBlock`x`newColsPerBlock`
+   * Even if all [[MatrixBlock]]s (except the ones on the right and bottom edge)
+   * have dimensions `newRowsPerBlock`x`newColsPerBlock`, a new [[BlockMatrix]] will be returned.
+   * The resulting number of partitions may be different than `suggestedNumPartitions`. Assumes
+   * that the offsets of each block is
+   * (`blockRowIndex`x`rowsPerBlock`, `blockColIndex`x`colsPerBlock`).
```

Comment:
Instead of repeating the docs, I think it would suffice to just specify they delegate to `repartition` with certain arguments set. Alternatively, instead of providing overloads we could just provide default argumetns.